packages/vats/src/virtual-purse.js

```diff
@@ -50,7 +51,9 @@ export const makeVirtualPurseKitIKit = (
   });
 
   const RetainRedeemI = M.interface('RetainRedeem', {
-    retain: M.callWhen(PaymentShape).optional(amountShape).returns(amountShape),
+    retain: M.callWhen(PaymentShape)
+      .optional(AmountShapeShape)
+      .returns(amountShape),
```

Comment:
**Upgrade Consideration**: That's a change to vat-bank. Are we planning to upgrade it?
If not, we might have to work around this instead.

Reply:
I dunno but it's a bug in the interface guard. Working around it means never using that parameter. 

If you're proposing to remove the parameter rather than fixing it (so nobody using a thing that will break in production) I can get behind that.

Reply:
Since all amounts are keys, and all keys are patterns, anything that passes the old incorrect `amountShape` guard (when that is the same as `AmountShape`) will pass either the new `M.pattern()` or `AmountShapeShape` guard, so this seems like a compatible change. It only turns correct-but-rejected arguments into correctly-accepted arguments.

Reply:
Yeah, upgrading this in production won't break any code that had managed to run in production before.

My remaining concern is that if we make this fix and don't ship it for a while then in the meantime people may right code that works in dev but will fail when they get to the production chain (or test in a3p). I'll take any changes out fo this PR and leave it for https://github.com/Agoric/agoric-sdk/issues/9407

Reply:
> ...people may right code that works in dev but will fail when they get to the production chain...

yes. that's exactly the upgrade consideration that I have in mind.

I'm proposing that until we have plans to deploy this fix, we don't put any code that depends on it into stuff that we plan to deploy.

> If you're proposing to remove the parameter rather than fixing it ...

I hadn't considered that. And I don't advocate it now (though I suppose I wouldn't object to it).

Reply:
See https://github.com/Agoric/agoric-sdk/pull/9410